dask_geopandas/core.py

```diff
@@ -307,6 +308,30 @@ def cx(self):
         return _CoordinateIndexer(self)
 
 
+    def hilbert_distance(self, p=15):
+
+        """
+        A function that calculates hilbert distance for each geometry
+        in each partition of a Dask-GeoDataFrame 
+        
+        Parameters
+        ----------    
+        p	: Hilbert curve parameter
+        
+        Returns
+        ----------
+        Distances for each partition
+        """
+        
+        # Compute total bounds of all partitions rather than each partition
+        total_bounds = self.total_bounds.compute()
```

Comment:
> I am curious if there's a way to do this lazily (not that I know one) so we don't have to call `compute` inside the method.

I realised this would raise a comment after pushing the code - It will work without calling compute. 

This was added by mistake when debugging a problem when applying the `_hilbert_distance` function to Dask gdfs. Initially I was calling total_bounds within `_hilbert_distance` , which resulted the computation of the total bounds of each partition - which created incorrect wrong.

Reply:
Simply removing `.compute()` fails on my end (`Metadata inference failed in lambda.`) but if you can get rid of it that would be great.

Reply:
Since we need to know the total bounds of all partitions before any single partition can start calculating the hilbert curve distances, I am not sure it can be avoided. 
(maybe the compute can be delayed to that dask does it later, but it will still require a full separate scan of the data).

Looking at spatialpandas, they don't call compute in [`_with_hilbert_distance_column`](https://github.com/holoviz/spatialpandas/blob/9252a7aba5f8bc7a435fffa2c31018af8d92942c/spatialpandas/dask.py#L508-L516), but that's because their [`total_bounds`](https://github.com/holoviz/spatialpandas/blob/9252a7aba5f8bc7a435fffa2c31018af8d92942c/spatialpandas/dask.py#L43-L50) attribute is not lazy, but directly returning values based on the bounds of the spatial partitions.

I am not sure we should do that as well for the `total_bounds` attribute, but at least here it would be good to check if spatial partitions are present, and if so use their total bounds (which will be much cheaper (not needing a full scan of the data) than actually calculating the full total_bounds, and be good enough for the use case here)

Reply:
> Since we need to know the total bounds of all partitions before any single partition can start calculating the hilbert curve distances, I am not sure it can be avoided.

This ^. We can discuss this tomorrow.

> Simply removing `.compute()` fails on my end (`Metadata inference failed in lambda.`) but if you can get rid of it that would be great.

You're right, maybe I was thinking off a previous step.

Reply:
updated in 0fba9a8